lib/download.go

```diff
@@ -1,47 +1,142 @@
 package lib
 
 import (
+	"errors"
 	"fmt"
 	"io"
+	"log"
 	"net/http"
 	"os"
 	"path/filepath"
 	"strings"
 )
 
-// DownloadFromURL : Downloads the binary from the source url
-func DownloadFromURL(installLocation string, url string) (string, error) {
+const (
+	pubKeyId     = "72D7468F"
+	pubKeyPrefix = "hashicorp_"
+	pubKeySuffix = ".asc"
+	pubKeyUri    = "https://www.hashicorp.com/.well-known/pgp-key.txt"
```

Comment:
Same point as for `pubKeyId` above: would it make sense to make it adjustable by users?
Additionally users may want to have their own mirror (or e.g. buildfarms w/o access to external internet) and even re-packed (built by themselves) binaries.

So it looks like it makes sense to allow verification to be 1) customizable and 2) switchable. What do you think?

Reply:
I just had a look into the possible configurations and i see a lot of consolidation potential there.
For example: the toml file only supports bin and version param (see #274), and i don't even see a test case for that (the test-data/test_tfswitchtoml/.tfswitch.toml) is unreferenced throughout the project.
i just started fiddling around with refactoring the parameter parsing but this is definitely out of scope of this ticket.

Reply:
Let me re-iterate and I'll leave this to you guys to decide then =)
https://github.com/warrensbox/terraform-switcher/pull/334#discussion_r1543158173 — we already were through the issue where `tfswitch` came up short for all of the users trying to download `terraform` because of the tiny change on the Hashicorp site end and users were absolutely blocked because they had no option to override the default. This is solely point that I frankly would not want to be introduced again.
While making this option customizable by users **looks to be** a scope of a separate ticket in general, I reckon the addition of a new functionality (which is the scope of this current PR) should better be accompanied with an option for users to override default values that configure this new functionality's behavior.
Thanks.

Reply:
@yermulnik i made the options for the public key configurable. Will you pleas have a look at the changes?

Reply:
That's for the extra context @yermulnik - I vaguely remember that now. 

I think there's definitely a bigger thing, as @MatrixCrawler says, around configuration options that we could look at in a separate item.

Reply:
@MatrixCrawler Looking. Would you mind resolving merge conflict meanwhile?

Reply:
I'be partially reviewed and I think we should not slap this and #350 together and prefer #350 to get merged first.